File: aat-android/src/main/java/ch/bailu/aat/views/BusyViewControlIID.java
package ch.bailu.aat.views;

import android.view.ViewGroup;

import ch.bailu.aat_lib.dispatcher.OnContentUpdatedInterface;
import ch.bailu.aat_lib.gpx.GpxInformation;

public class BusyViewControlIID extends BusyViewControl implements OnContentUpdatedInterface {

    public BusyViewControlIID(ViewGroup parent) {
        super(parent);
    }


    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        if (info.isLoaded())
            stopWaiting(iid);

        else
            startWaiting(iid);

    }


}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/busy/BusyViewControlIID.kt
package ch.bailu.aat.views.busy

import android.view.ViewGroup
import ch.bailu.aat_lib.dispatcher.TargetInterface
import ch.bailu.aat_lib.gpx.GpxInformation

open class BusyViewControlIID(parent: ViewGroup) : BusyViewControl(parent), TargetInterface {
    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        if (info.getLoaded()) stopWaiting(iid) else startWaiting(iid)
    }
}
